File: movie-storage/src/main/java/com/movie/moviestorage/movie/entity/SeatEntity.java
package com.movie.moviestorage.movie.entity;

import com.movie.moviestorage.BaseEntity;
import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.Objects;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Table(name = "seat")
public class SeatEntity extends BaseEntity {

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(columnDefinition = "BIGINT UNSIGNED")
    private Long id;

    private Long screenId;

    @Column(columnDefinition = "VARCHAR(10) NOT NULL COMMENT '좌석번호'")
    private String seatNumber;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof SeatEntity seatEntity)) return false;
        return id != null && id.equals(seatEntity.id);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id);
    }

}

File: movie/infra/src/main/kotlin/com/example/movie/persistence/seat/entity/SeatEntity.kt
package com.example.movie.persistence.seat.entity

import com.example.movie.domain.seat.model.Seat
import com.example.movie.persistence.common.BaseEntity
import com.example.movie.persistence.theater.entity.TheaterEntity
import jakarta.persistence.*

@Entity
@Table(name = "seat")
class SeatEntity(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "seat_id", columnDefinition = "INT UNSIGNED")
    val id: Long = 0,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "theater_id", columnDefinition = "INT UNSIGNED", 
        foreignKey = ForeignKey(ConstraintMode.NO_CONSTRAINT))
    val theater: TheaterEntity,

    @Column(name = "row_val", nullable = false)
    val rowVal: Char,

    @Column(name = "col_val", nullable = false)
    val colVal: Int
) : BaseEntity() {
    companion object {
        fun from(seat: Seat) =SeatEntity(
            id = seat.id,
            theater = TheaterEntity.from(seat.theater),
            rowVal = seat.row,
            colVal = seat.col,
        ).apply {
            createdBy = theater.createdBy
            createdAt = seat.createdAt
            updatedBy = theater.updatedBy
            updatedAt = seat.updatedAt
        }
    }
    fun toDomain(): Seat {
        return Seat(
            id = id,
            theater = theater.toDomain(),
            row = rowVal,
            col = colVal,
            createdBy = createdBy,
            createdAt = createdAt,
            updatedBy = updatedBy,
            updatedAt = updatedAt
        )
    }
}